lib/utils/fields.js

```diff
@@ -89,12 +83,48 @@ export function dataObjectToArray(dataObject, fields, options = {}) {
     }
     const field = fields[fieldIdx];
     if (field.multivalued && Array.isArray(value)) {
-      dataArray[fieldIdx] = value
-        .map((v) => stringifyDatatype(v, field.datatype))
-        .join(multivalueDelimiter);
+      dataArray[fieldIdx] = formatMultivaluedValue(
+        value.map((v) => stringifyDatatype(v, field.datatype))
+      );
     } else {
       dataArray[fieldIdx] = stringifyDatatype(value, field.datatype);
     }
   }
   return dataArray;
 }
+
+/**
+ * Parse a formatted string representing a multivalued value and return an
+ * array of the individual values.
+ *
+ * @param {String} value String-formatted multivalued value.
+ * @return {Array<String>} Array of individual string values.
+ */
+export function parseMultivaluedValue(value) {
+  if (!value) {
+    return [];
+  }
+  // trim the delimiter and the resulting tokens to be flexible about what
+  // this function accepts
+  return value
+    .split(MULTIVALUED_DELIMITER.trim())
+    .map((v) => v.trim())
+    .filter((v) => !!v);
```

Comment:
Just curious, does ".filter((v) => !!v);" remove null values?

Reply:
It filters out all [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) by casting the value to a boolean (`!v`) and then inverting the result (`!!v`). So yeah here it's being used to filter out nulls and empty strings.